include/oneapi/dpl/internal/random_impl/uniform_real_distribution.h

```diff
@@ -195,7 +195,7 @@ class uniform_real_distribution
         sycl::vec<scalar_type, _Ndistr> __res;
         int __i;
         constexpr int __tail_size = _Ndistr % _Negnine;
-        for (__i = 0; __i < _Ndistr; __i += _Negnine)
+        for (__i = 0; __i < _Ndistr - __tail_size; __i += _Negnine)
```

Comment:
Why __tail_size is removed here?

Reply:
Discussed offline

Reply:
Could you please summarize what was discussed offline? The reason is not obvious for me as well

Reply:
This code branch is used when _Ndistr > _Nengine and the loop is aimed to process part before tail, that's why the upper limit is _Ndistr - __tail_size. There is another loop to process last _tail_size elements. Previously we got incorrect results for _Ndistr = 3, _Nengine = 2, because we had extra iterations inside the loop (we processed 4 numbers instead of 2) and state of the engine was changed due to this, so when we were processing tail we were producing incorrect number (overwriting 3rd element)
Pavel accidentally thought that I've changed it from _Ndistr - __tail_size to _Ndistr, but the changes are opposite